### PR TITLE
Support windows with `juv run`

### PR DESCRIPTION
**Pull request by ATL2001**
I only had to make a few code tweaks to get the code working on my windows box for firing up the subprocess, and I added raw string syntax for the notebook path in case the notebook's path has a \t in it so it doesnt change the \t into a tab. or a \u into a unicode escape etc.  and I added `ignore_cleanup_errors=True` to all the tempdirectory usage because without it, an error would throw anytime I shutdown jupyterlab because the nbsignatures.db of jupyter was locked.

To get the tests to pass I had to replace the pipes at the end of some of the lines with the pipe symbol from my keyboard (I'm not going to pretend to be an encoding expert, but when I did a find/replace on the pipe symbols in the existing code with the pipe symbol on my keyboard, magically the tests passed for me)

let me know what you think!

### CODE REVIEW

File: src/juv/_run_replace.py [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
 
 import os
+import platform
 import signal
 import subprocess
 import sys
 
 from uv import find_uv_bin
 
+IS_WINDOWS = platform.system()
```
Comment (manzt) [inline src/juv/_run_replace.py:11]: Could we be more explicit here? I wonder about:

```suggestion
IS_WINDOWS = sys.platform.startswith("win")
```

and then remove the unecessary `platform` import.

File: src/juv/_run_template.py [hunk 2]
```diff
@@ -75,7 +75,7 @@ def as_with_arg(self) -> str:
 juv_data_dir = Path(user_data_dir("juv"))
 juv_data_dir.mkdir(parents=True, exist_ok=True)
 
-temp_dir = tempfile.TemporaryDirectory(dir=juv_data_dir)
+temp_dir = tempfile.TemporaryDirectory(dir=juv_data_dir, ignore_cleanup_errors=True)
```
Comment (manzt) [inline src/juv/_run_template.py:78]: It looks like `ignore_cleanup_errors` [was added](https://github.com/python/cpython/pull/24793) in Python 3.10, so some of the tests are failing.

I'd like to support whatever is not EOL, so maybe we could just make a little subclass with the behavior we want:

```python
# Custom TemporaryDirectory for Python < 3.10
# TODO: Use `ignore_cleanup_errors=True` in Python 3.10+
class TemporaryDirectoryIgnoreErrors(tempfile.TemporaryDirectory):
    def cleanup(self):
        try:
            super().cleanup()
        except Exception:
            pass  # Ignore cleanup errors
```

and just copy that code snippet into the two files where it is used.

File: src/juv/_run_template.py [hunk 3]
```diff
@@ -167,7 +167,7 @@ def handle_termination(signum, frame):
     version = importlib.metadata.version("notebook")
     print("JUV_MANGED=" + "notebook" + "," + version, file=sys.stderr)
 
-sys.argv = ["jupyter-notebook", "{notebook}", *{args}]
+sys.argv = ["jupyter-notebook", r"{notebook}", *{args}]
```
Comment (manzt) [inline src/juv/_run_template.py:170]: thanks!!

File: tests/test_juv.py [hunk 4]
```diff
@@ -851,11 +855,11 @@ def test_stamp_script(
...
+def main() -> None:                                                                                                                                                                                                               |
+    print("Hello from foo.py!")                                                                                                                                                                                                   |
+                                                                                                                                                                                                                                  |
+                                                                                                                                                                                                                                  |
+if __name__ == "__main__":                                                                                                                                                                                                        |
```
Comment (manzt) [inline tests/test_juv.py:862]: For these snippets, I wonder about setting `encoding='utf-8'` within `Path.open`. E..g.,

```py
        with (tmp_path / "foo.py").open("w", encoding="utf-8") as f:
```

I would prefer to not have the pipes but am also not an encoding expert. The tests still pass for me as is, and I can fiddle around with this myself too so no need to spend a lot of time on it.

**Review by manzt — COMMENTED** [review]
Thanks for the awesome contribution! Sorry it took me so long for the review-preparing for my phd defense next week.

Just a couple of comments that hopefully we can get addressed quickly! Thanks again.

File: src/juv/_run_replace.py (hunk 1 above)
Comment (ATL2001) [reply src/juv/_run_replace.py:11]: oh neat! I didn't realize sys had that functionality... and what I had there wasn't doing what I wanted anyway, oops, that's totally on me! I intended to put `platform.system=="Windows"`  :-/ I like your suggestion better either way!

**Comment by ATL2001** [comment]
I thought I had run ruff format before committing stuff, but I'm sure I did this time!  good luck on the defense! 👍

**Review by manzt — COMMENTED** [review]
Thanks for the awesome contribution!

**Review by manzt — APPROVED** [review]
Thanks for the awesome contribution!

**Comment by ATL2001** [comment]
happy to help! thanks for making it in the first place! I can't wait to show it off to some people, do you have any idea when the windows stuff will be available from pypi? 
- Aaron

**Comment by manzt** [comment]
Shipped in [`v0.2.26`](https://github.com/manzt/juv/releases/tag/v0.2.26)! Thanks

**Comment by manzt** [comment]
Whoops, there was a bug where `TemporaryDirectory` in the templated didn't get updated. Fixed in v0.2.27.

**Pull request merged**